### PR TITLE
br/pkg/utils: migrate tests to testify

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/28183

Problem Summary:

Migrate tests to testify for `br/pkg/utils`.

### What is changed and how it works?

Migrate tests to testify for `br/pkg/utils`. 
Let `ProgressPrinter.Close` wait background goroutine to exit.

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/utils/key_test.go [hunk 1]
```diff
@@ -89,11 +88,13 @@ func (r *testKeySuite) TestParseKey(c *C) {
 
 	for _, tt := range testNotSupportKey {
 		_, err := ParseKey("notSupport", tt.any)
-		c.Assert(err, ErrorMatches, "unknown format.*")
+		require.Regexp(t, "unknown format.*", err.Error())
```
Comment (kennytm) [inline br/pkg/utils/key_test.go:91]:
```suggestion
		require.Error(t, err)
		require.Regexp(t, "unknown format.*", err.Error())
```

File: br/pkg/utils/progress_test.go [hunk 2]
```diff
@@ -34,15 +33,16 @@ func (r *testProgressSuite) TestProgress(c *C) {
 	progress2.Inc()
 	time.Sleep(2 * time.Second)
 	p = <-pCh2
-	c.Assert(p, Matches, `.*"P":"50\.00%".*`)
+	require.Regexp(t, `.*"P":"50\.00%".*`, p)
```
Comment (kennytm) [inline br/pkg/utils/progress_test.go:36]:
```suggestion
	require.Contains(t, p, `"P":"50.00%"`)
```

File: br/pkg/utils/progress_test.go [hunk 3]
```diff
@@ -68,10 +68,11 @@ func (r *testProgressSuite) TestProgress(c *C) {
...
 	time.Sleep(2 * time.Second)
 	p = <-pCh8
-	c.Assert(p, Matches, `.*"P":"25\.00%".*`)
+	require.Regexp(t, `.*"P":"25\.00%".*`, p)
 
 	// Cancel should stop progress at the current position.
 	cancel()
 	p = <-pCh8
-	c.Assert(p, Matches, `.*"P":"25\.00%".*`)
+	require.Regexp(t, `.*"P":"25\.00%".*`, p)
```
Comment (kennytm) [inline br/pkg/utils/progress_test.go:76]: ... (ditto for everything above)

File: br/pkg/utils/safe_point_test.go [hunk 4]
```diff
@@ -5,41 +5,33 @@ package utils_test
...
 	}
 	{
 		err := utils.CheckGCSafePoint(ctx, pdClient, 2333-1)
-		c.Assert(err, NotNil)
+		require.NotNil(t, err)
 	}
 	{
 		err := utils.CheckGCSafePoint(ctx, pdClient, 0)
-		c.Assert(err, ErrorMatches, ".*GC safepoint 2333 exceed TS 0.*")
+		require.Regexp(t, ".*GC safepoint 2333 exceed TS 0.*", err)
```
Comment (kennytm) [inline br/pkg/utils/safe_point_test.go:34]:
```suggestion
		require.Error(t, err)
		require.Contains(t, err.Error(), "GC safepoint 2333 exceed TS 0")
```

File: br/pkg/utils/safe_point_test.go [hunk 5]
```diff
@@ -5,41 +5,33 @@ package utils_test
...
 	pdClient := &mockSafePoint{safepoint: 2333}
 	{
 		err := utils.CheckGCSafePoint(ctx, pdClient, 2333+1)
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 	}
 	{
 		err := utils.CheckGCSafePoint(ctx, pdClient, 2333)
-		c.Assert(err, NotNil)
+		require.NotNil(t, err)
```
Comment (kennytm) [inline br/pkg/utils/safe_point_test.go:26]:
```suggestion
		require.Error(t, err)
```

File: br/pkg/utils/safe_point_test.go [hunk 6]
```diff
@@ -5,41 +5,33 @@ package utils_test
...
 	}
 	{
 		err := utils.CheckGCSafePoint(ctx, pdClient, 2333-1)
-		c.Assert(err, NotNil)
+		require.NotNil(t, err)
```
Comment (kennytm) [inline br/pkg/utils/safe_point_test.go:30]:
```suggestion
		require.Error(t, err)
```

File: br/pkg/utils/safe_point_test.go [hunk 7]
```diff
@@ -128,11 +122,11 @@ func (s *testSafePointSuite) TestStartServiceSafePointKeeper(c *C) {
 	for i, cs := range cases {
 		ctx, cancel := context.WithCancel(context.Background())
 		err := utils.StartServiceSafePointKeeper(ctx, pdClient, cs.sp)
-		checker := IsNil
-		if !cs.ok {
-			checker = NotNil
+		if cs.ok {
+			require.NoErrorf(t, err, "case #%d, %v", i, cs)
+		} else {
+			require.NotNilf(t, err, "case #%d, %v", i, cs)
```
Comment (kennytm) [inline br/pkg/utils/safe_point_test.go:128]:
```suggestion
			require.Errorf(t, err, "case #%d, %v", i, cs)
```

File: br/pkg/utils/schema_test.go [hunk 8]
```diff
@@ -89,28 +86,43 @@ func (r *testSchemaSuite) TestLoadBackupMeta(c *C) {
...
 			&backuppb.CipherInfo{
 				CipherType: encryptionpb.EncryptionMethod_PLAINTEXT,
 			}),
 	)
 	tbl := dbs[dbName.String()].GetTable(tblName.String())
-	c.Assert(err, IsNil)
-	c.Assert(tbl.Files, HasLen, 1)
-	c.Assert(tbl.Files[0].Name, Equals, "1.sst")
+	require.NoError(t, err)
+	require.Equal(t, 1, len(tbl.Files))
```
Comment (kennytm) [inline br/pkg/utils/schema_test.go:106]:
```suggestion
	require.Len(t, tbl.Files, 1)
```

File: br/pkg/utils/schema_test.go [hunk 9]
```diff
@@ -180,25 +192,25 @@ func (r *testSchemaSuite) TestLoadBackupMetaPartionTable(c *C) {
...
 			&backuppb.CipherInfo{
 				CipherType: encryptionpb.EncryptionMethod_PLAINTEXT,
 			},
 		),
 	)
 	tbl := dbs[dbName.String()].GetTable(tblName.String())
-	c.Assert(err, IsNil)
-	c.Assert(tbl.Files, HasLen, 3)
+	require.NoError(t, err)
+	require.Equal(t, 3, len(tbl.Files))
```
Comment (kennytm) [inline br/pkg/utils/schema_test.go:213]:
```suggestion
	require.Len(t, tbl.Files, 3)
```

File: br/pkg/utils/schema_test.go [hunk 10]
```diff
@@ -257,87 +269,95 @@ func buildBenchmarkBackupmeta(c *C, dbName string, tableCount, fileCountPerTable
...
 			),
 		)
-		c.Assert(err, IsNil)
-		c.Assert(dbs, HasLen, 1)
-		c.Assert(dbs, HasKey, "bench")
-		c.Assert(dbs["bench"].Tables, HasLen, 64)
+		require.NoError(b, err)
+		require.Equal(b, 1, len(dbs))
+		require.Contains(b, dbs, "bench")
+		require.Equal(b, 64, len(dbs["bench"].Tables))
```
Comment (kennytm) [inline br/pkg/utils/schema_test.go:299]: ditto

File: br/pkg/utils/schema_test.go [hunk 11]
```diff
@@ -257,87 +269,95 @@ func buildBenchmarkBackupmeta(c *C, dbName string, tableCount, fileCountPerTable
...
 			),
 		)
-		c.Assert(err, IsNil)
-		c.Assert(dbs, HasLen, 1)
-		c.Assert(dbs, HasKey, "bench")
-		c.Assert(dbs["bench"].Tables, HasLen, 1024)
+		require.NoError(b, err)
+		require.Equal(b, 1, len(dbs))
+		require.Contains(b, dbs, "bench")
+		require.Equal(b, 1024, len(dbs["bench"].Tables))
```
Comment (kennytm) [inline br/pkg/utils/schema_test.go:330]: ditto

File: br/pkg/utils/schema_test.go [hunk 12]
```diff
@@ -257,87 +269,95 @@ func buildBenchmarkBackupmeta(c *C, dbName string, tableCount, fileCountPerTable
...
 			),
 		)
-		c.Assert(err, IsNil)
-		c.Assert(dbs, HasLen, 1)
-		c.Assert(dbs, HasKey, "bench")
-		c.Assert(dbs["bench"].Tables, HasLen, 10240)
+		require.NoError(b, err)
+		require.Equal(b, 1, len(dbs))
+		require.Contains(b, dbs, "bench")
+		require.Equal(b, 10240, len(dbs["bench"].Tables))
```
Comment (kennytm) [inline br/pkg/utils/schema_test.go:361]: ditto

File: br/pkg/utils/env_test.go [hunk 13]
```diff
@@ -25,20 +24,20 @@ func (s *envSuit) TestProxyFields(c *check.C) {
...
 			}
 		}
 
 		for _, field := range proxyFields() {
 			idx, ok := revIndex[field.Key]
-			c.Assert(ok, check.IsTrue)
-			c.Assert((1<<idx)&mask, check.Not(check.Equals), 0)
-			c.Assert(field.String, check.Equals, envPreset[idx])
+			require.True(t, ok)
+			require.NotEqual(t, 0, (1<<idx)&mask)
```
Comment (kennytm) [inline br/pkg/utils/env_test.go:39]:
```suggestion
			require.NotZero(t, (1<<idx)&mask)
```

File: br/pkg/utils/json_test.go [hunk 14]
```diff
@@ -212,40 +207,19 @@ var testMetaJSONs = [][]byte{
...
 		metaJSON, err := MarshalBackupMeta(meta)
-		c.Assert(err, IsNil)
-		c.Assert(metaJSON, jsonEquals{}, testMetaJSON)
+		require.NoError(t, err)
+
+		var metaMap map[string]interface{}
+		require.NoError(t, json.Unmarshal(metaJSON, &metaMap))
+		var testMetaMap map[string]interface{}
+		require.NoError(t, json.Unmarshal(testMetaJSON, &testMetaMap))
+		require.Equal(t, testMetaMap, metaMap)
```
Comment (kennytm) [inline br/pkg/utils/json_test.go:223]:
```suggestion
		require.JSONEq(t, string(testMetaJSON), string(metaJSON))
```

File: br/pkg/utils/schema_test.go [hunk 15]
```diff
@@ -6,39 +6,36 @@ import (
...
 	}
 }
 
-func (r *testSchemaSuite) TestLoadBackupMeta(c *C) {
+func TestLoadBackupMeta(t *testing.T) {
+	t.Parallel()
+
+	testDir, err := os.MkdirTemp("", "load_backup_meta_test")
+	require.NoError(t, err)
+	defer os.RemoveAll(testDir)
```
Comment (kennytm) [inline br/pkg/utils/schema_test.go:35]:
```suggestion
	testDir := t.TempDir()
```

https://pkg.go.dev/testing#T.TempDir

File: br/pkg/utils/schema_test.go [hunk 16]
```diff
@@ -89,28 +86,43 @@ func (r *testSchemaSuite) TestLoadBackupMeta(c *C) {
...
 
-func (r *testSchemaSuite) TestLoadBackupMetaPartionTable(c *C) {
+func newStore(t require.TestingT) (store storage.ExternalStorage, cleanFunc func()) {
+	base, err := os.MkdirTemp("", "br_schema_test")
+	require.NoError(t, err)
+	store, err = storage.NewLocalStorage(base)
+	require.NoError(t, err)
+	return store, func() {
+		require.NoError(t, os.RemoveAll(base))
+	}
```
Comment (kennytm) [inline br/pkg/utils/schema_test.go:117]: ditto

**Review by kennytm — COMMENTED** [review]
rest LGTM

File: br/pkg/utils/schema_test.go [hunk 17]
```diff
@@ -107,21 +105,12 @@ func TestLoadBackupMeta(t *testing.T) {
...
-		require.NoError(t, os.RemoveAll(base))
-	}
-}
-
 func TestLoadBackupMetaPartionTable(t *testing.T) {
 	t.Parallel()
 
-	store, cleanFunc := newStore(t)
-	defer cleanFunc()
+	testDir := os.TempDir()
```
Comment (kennytm) [inline br/pkg/utils/schema_test.go:111]:
```suggestion
	testDir := t.TempDir()
```

Comment (gozssky) [reply br/pkg/utils/schema_test.go]: 😂

**Review by kennytm — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

**Pull request merged**